include/xframe/xmasked_view.hpp

```diff
@@ -0,0 +1,581 @@
+        using inner_backstrides_type = typename value_expression::inner_backstrides_type;
+
+        using expression_tag = xoptional_expression_tag;
+
+        using iterable_base = xiterable<xmasked_view<CTD, CTM>>;
+        using stepper = typename iterable_base::stepper;
+        using const_stepper = typename iterable_base::const_stepper;
+
+        template <layout_type L>
+        using layout_iterator = typename iterable_base::template layout_iterator<L>;
+        template <layout_type L>
+        using const_layout_iterator = typename iterable_base::template const_layout_iterator<L>;
+        template <layout_type L>
+        using reverse_layout_iterator = typename iterable_base::template reverse_layout_iterator<L>;
+        template <layout_type L>
+        using const_reverse_layout_iterator = typename iterable_base::template const_reverse_layout_iterator<L>;
+
+        template <class S, layout_type L>
+        using broadcast_iterator = typename iterable_base::template broadcast_iterator<S, L>;
+        template <class S, layout_type L>
+        using const_broadcast_iterator = typename iterable_base::template const_broadcast_iterator<S, L>;
+        template <class S, layout_type L>
+        using reverse_broadcast_iterator = typename iterable_base::template reverse_broadcast_iterator<S, L>;
+        template <class S, layout_type L>
+        using const_reverse_broadcast_iterator = typename iterable_base::template const_reverse_broadcast_iterator<S, L>;
+
+        using iterator = typename iterable_base::iterator;
+        using const_iterator = typename iterable_base::const_iterator;
+        using reverse_iterator = typename iterable_base::reverse_iterator;
+        using const_reverse_iterator = typename iterable_base::const_reverse_iterator;
+
+        template <class D, class M>
+        xmasked_view(D&& data, M&& mask);
+
+        size_type size() const noexcept;
+        constexpr size_type dimension() const noexcept;
+
+        const inner_shape_type& shape() const noexcept;
+        const inner_strides_type& strides() const noexcept;
+        const inner_backstrides_type& backstrides() const noexcept;
+
+        layout_type layout() const noexcept;
+
+        template <class T>
+        void fill(const T& value);
+
+        template <class... Args>
+        reference operator()(Args... args);
+
+        template <class... Args>
+        const_reference operator()(Args... args) const;
+
+        template <class... Args>
+        reference at(Args... args);
+
+        template <class... Args>
+        const_reference at(Args... args) const;
+
+        template <class... Args>
+        reference unchecked(Args... args);
+
+        template <class... Args>
+        const_reference unchecked(Args... args) const;
+
+        template <class S>
+        disable_integral_t<S, reference> operator[](const S& index);
+        template <class I>
+        reference operator[](std::initializer_list<I> index);
```

Comment:
So you want me to remove it completely ? Isn't it a breaking change in xtensor ?

Reply:
actually we need to define what we make it i guess.
I think it should take an index_type (i.e. std::array<std::size_t, ... > or svector<size_t>) then we can still easily construct it from an initializer list.

we need to discuss this (anyways, since it is implemented like this in xtensor, we can prob keep it for now :)

Reply:
Ok I see :) maybe we should open an issue in xtensor ?